web3/_utils/contracts.py

```diff
@@ -240,6 +241,8 @@ def get_function_info(fn_name, contract_abi=None, fn_abi=None, args=None, kwargs
 
     fn_arguments = merge_args_and_kwargs(fn_abi, args, kwargs)
 
+    _, fn_arguments = get_abi_inputs(fn_abi, fn_arguments)
```

Comment:
*nit* overwrote `fn_arguments`, but now it has a slightly different meaning. Maybe `typed_arguments` or `annotated_arguments`? (same as the one in `web3/_utils/abi` I commented on)